### PR TITLE
Move the SecondaryHierarchy to its own file in IBTK.

### PR DESCRIPTION
**Pull request by drwells**
Yet another part of #1235. We're getting closer - the main things left now are adding some more accessor functions and then cleaning up the load balancing logic (which will put a big dent in #1126, since we will avoid calculating workloads two more times).

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: ibtk/include/ibtk/SecondaryHierarchy.h [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace IBTK
+{
+/**
+ * Class encapsulating the idea of a secondary hierarchy. In practice, this
+ * is an equivalent (same index space) hierarchy to a given PatchHierarchy,
```
Comment (boyceg) [inline ibtk/include/ibtk/SecondaryHierarchy.h:46]: Is it equivalent in the sense that corresponding levels cover the same region?

**Review by boyceg — APPROVED** [review]

File: ibtk/include/ibtk/SecondaryHierarchy.h (hunk 1 above)
Comment (drwells) [reply ibtk/include/ibtk/SecondaryHierarchy.h:46]: yup!

Comment (drwells) [reply ibtk/include/ibtk/SecondaryHierarchy.h:46]: My intent with that statement was to state that this should be the same patches in a different order.

Comment (boyceg) [reply ibtk/include/ibtk/SecondaryHierarchy.h:46]: Is that what we actually want foe the typical use case?

**Pull request merged**